### PR TITLE
Require Node v18 in Wrangler v4

### PR DESCRIPTION
**Pull request by penalosa**
Bump minimum supported Node version to Node v18 in Wrangler v4

---

- Tests
  - [ ] TODO (before merge)
  - [x] Tests included
  - [ ] Tests not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [x] Required
  - [ ] Not required because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: will be part of v4 docs

### CODE REVIEW

File: .github/workflows/test-and-check.yml [hunk 1]
```diff
@@ -102,10 +102,21 @@ jobs:
       matrix:
         os: [ubuntu-latest, windows-latest, macos-13]
         filter: ["./packages/*", "./fixtures/*"]
+        node_version: [18]
```
Comment (emily-shen) [inline .github/workflows/test-and-check.yml:105]: I think CI is pulling something like v18.20.2 which is too old - might need to make these more specific.

**Review by andyjessop — APPROVED** [review]

**Comment by emily-shen** [comment]
also there's a line in the c3 nuxt templates that might need to be updated?
`writeFile("./.node-version", "17");`

File: .github/workflows/test-and-check.yml (hunk 1 above)
Comment (penalosa) [reply .github/workflows/test-and-check.yml:105]: I've loosened this up a bit to only require v18

**Comment by penalosa** [comment]
> also there's a line in the c3 nuxt templates that might need to be updated? `writeFile("./.node-version", "17");`

I think we can fix this without a major, so I'll open a separate PR

**Pull request merged**